client/commands/v2/pysa_server.py

```diff
@@ -100,231 +136,13 @@ async def log_and_show_message_to_client(
-                        f"Cannot start a new Pyre server at `{self.server_identifier}`.",
-                        level=lsp.MessageType.ERROR,
-                    )
-
-                    if (
-                        self.server_state.consecutive_start_failure
-                        == CONSECUTIVE_START_ATTEMPT_THRESHOLD - 1
-                    ):
-                        # The heuristic here is that if the restart has failed
-                        # this many times, it is highly likely that the restart was
-                        # blocked by a defunct socket file instead of a concurrent
-                        # server start, in which case removing the file could unblock
-                        # the server.
-                        LOG.warning(f"Removing defunct socket file {socket_path}...")
-                        stop.remove_socket_if_exists(socket_path)
-                else:
-                    await self.show_message_to_client(
-                        (
-                            f"Pyre server restart at `{self.server_identifier}` has been "
-                            + "failing repeatedly. Disabling The Pyre plugin for now."
-                        ),
-                        level=lsp.MessageType.ERROR,
-                    )
-                    _log_lsp_event(
-                        remote_logging=self.pyre_arguments.remote_logging,
-                        event=LSPEvent.SUSPENDED,
-                        normals=self._auxiliary_logging_info(),
-                    )
-
-            else:
-                raise RuntimeError("Impossible type for `start_status`")
-
-    async def run(self) -> None:
-        try:
-            await self._run()
-        except Exception:
-            _log_lsp_event(
-                remote_logging=self.pyre_arguments.remote_logging,
-                event=LSPEvent.DISCONNECTED,
-                normals={
-                    **self._auxiliary_logging_info(),
-                    "exception": traceback.format_exc(),
-                },
-            )
-            raise
-
-
-class PysaServer:
-    # I/O Channels
-    input_channel: connection.TextReader
-    output_channel: connection.TextWriter
-
-    # Immutable States
-    client_capabilities: lsp.ClientCapabilities
-
-    # Mutable States
-    state: ServerState
-    pyre_manager: connection.BackgroundTaskManager
-
-    def __init__(
-        self,
-        input_channel: connection.TextReader,
-        output_channel: connection.TextWriter,
-        client_capabilities: lsp.ClientCapabilities,
-        state: ServerState,
-        pyre_manager: connection.BackgroundTaskManager,
-    ) -> None:
-        self.input_channel = input_channel
-        self.output_channel = output_channel
-        self.client_capabilities = client_capabilities
-        self.state = state
-        self.pyre_manager = pyre_manager
+        for path, diagnostic in diagnostics.items():
+            await _publish_diagnostics(self.output_channel, path, [])
```

Comment:
Couldn't we delete the next two lines and just do this?
```suggestion
            await _publish_diagnostics(self.output_channel, path, diagnostic or [])
```

Reply:
Oh yeah, just changed and tested it, it works like this